addons/hr/models/res_users.py

```diff
@@ -116,7 +116,7 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
         # However, in this case, we want the user to be able to read/write its own data,
         # even if they are protected by groups.
         # We make the front-end aware of those fields by sending all field definitions.
-        if not self.env.user.share:
+        if not self.env.user.share and self._context.get('res_users_my_profile_mode'):
```

Comment:
possibly simpler version, without needing the XML change:
```suggestion
        profile_view = self.env.ref("hr.res_users_view_form_profile")
        if profile_view and view_id == profile_view.id:
```

Reply:
and also one can pass this `context` in RPC call to bypass this.